spec/Helpers.coffee

```diff
@@ -643,61 +684,140 @@ describe 'Component traits', ->
             , req
 
     describe 'when there are multiple output routes', ->
-      c = new noflo.Component
-      c.inPorts.add 'num', datatype: 'int'
-      .add 'str', datatype: 'string'
-      c.outPorts.add 'odd', datatype: 'object'
-      .add 'even', datatype: 'object'
-      num = new noflo.internalSocket.createSocket()
-      str = new noflo.internalSocket.createSocket()
-      odd = new noflo.internalSocket.createSocket()
-      even = new noflo.internalSocket.createSocket()
-      c.inPorts.num.attach num
-      c.inPorts.str.attach str
-      c.outPorts.odd.attach odd
-      c.outPorts.even.attach even
-
       it 'should send output to one or more of them', (done) ->
         numbers = ['cero', 'uno', 'dos', 'tres', 'cuatro', 'cinco', 'seis', 'siete', 'ocho', 'nueve']
+        c = new noflo.Component
+        c.inPorts.add 'num', datatype: 'int'
+        .add 'str', datatype: 'string'
+        c.outPorts.add 'odd', datatype: 'object'
+        .add 'even', datatype: 'object'
+        num = new noflo.internalSocket.createSocket()
+        str = new noflo.internalSocket.createSocket()
+        odd = new noflo.internalSocket.createSocket()
+        even = new noflo.internalSocket.createSocket()
+        c.inPorts.num.attach num
+        c.inPorts.str.attach str
+        c.outPorts.odd.attach odd
+        c.outPorts.even.attach even
 
         noflo.helpers.WirePattern c,
           in: ['num', 'str']
           out: ['odd', 'even']
           async: true
           ordered: true
+          forwardGroups: true
         , (data, groups, outs, callback) ->
           setTimeout ->
             if data.num % 2 is 1
-              outs.odd.beginGroup grp for grp in groups
```

Comment:
Yep, `groups` is a bit finicky because in Process API it is resolved properly only when dealing with output queue, so after the processing callback has fired.